tracer/src/Datadog.Trace/ClrProfiler/Instrumentation.cs

```diff
@@ -60,6 +61,62 @@ public static class Instrumentation
         [MethodImpl(MethodImplOptions.NoInlining)]
         public static string GetNativeTracerVersion() => "None";
 
+        private static void PropagateStableConfiguration()
+        {
+            // TODO: only for profiler today
+            //
+
+            // profiler is not available on ARM(64) don't even try to call the p/invoke
+            // as the binary won't be there
+            var fd = FrameworkDescription.Instance;
+            if ((fd.ProcessArchitecture == ProcessArchitecture.Arm64) || (fd.ProcessArchitecture == ProcessArchitecture.Arm))
+            {
+                Log.Information("Profiling is not available on ARM.");
+                return;
+            }
+
+            // it is also possible that the profiler binary is not "there" such as Azure Function and some CI tests
+            if (string.IsNullOrEmpty(Environment.GetEnvironmentVariable("DD_INTERNAL_PROFILING_NATIVE_ENGINE_PATH")))
+            {
+                Log.Information("Profiling native library path is not set, skipping setting profiler configuration.");
+                return;
+            }
+
+            Log.Debug("Setting Stable Configuration in Continuous Profiler native library.");
+            var tracerSettings = Tracer.Instance.Settings;
+            var profilerSettings = Profiler.Instance.Settings;
+
+            NativeInterop.SharedConfig config = new NativeInterop.SharedConfig
+            {
+                ProfilingEnabled = profilerSettings.ProfilerState switch
+                {
+                    ProfilerState.Auto => NativeInterop.ProfilingEnabled.Auto,
+                    ProfilerState.Enabled => NativeInterop.ProfilingEnabled.Enabled,
+                    _ => NativeInterop.ProfilingEnabled.Disabled
+                },
+
+                TracingEnabled = tracerSettings.TraceEnabled,
+                IastEnabled = Iast.Iast.Instance.Settings.Enabled,
+                RaspEnabled = Security.Instance.Settings.RaspEnabled,
+                DynamicInstrumentationEnabled = false,  // TODO: find where to get this value from but for the other native p/invoke call
+                RuntimeId = RuntimeId.Get(),
+                Environment = tracerSettings.Environment,
+                ServiceName = TraceUtil.NormalizeTag(tracerSettings.ServiceName ?? string.Empty),
```

Comment:
This is using the wrong value, you need to use `tracer.DefaultServiceName`:
```suggestion
                ServiceName = TraceUtil.NormalizeTag(tracer.DefaultServiceName),
```